imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java

```diff
@@ -1079,7 +1080,21 @@ public void run() {
                 break;
             case CLICKED:
                 isAdIsPaused = true;
-                messageBus.post(new AdEvent(AdEvent.Type.CLICKED));
+                Ad ad = adEvent.getAd();
+
+                try {
+                    Method clickThroughMethod = ad.getClass().getMethod("getClickThruUrl");
+                    String clickThruUrl = (String) clickThroughMethod.invoke(ad);
+                    messageBus.post(new AdEvent.AdClickedEvent(clickThruUrl));
+                    return;
+                } catch (NoSuchMethodException e) {
+                    e.printStackTrace();
+                } catch (IllegalAccessException e) {
+                    e.printStackTrace();
+                } catch (InvocationTargetException e) {
+                    e.printStackTrace();
+                }
+                messageBus.post(new AdEvent.AdClickedEvent("Unavailable"));
```

Comment:
Better use `null` instead of `"Unavailable"`.

Reply:
ok